internal/semver/compare.go

```diff
@@ -6,150 +6,43 @@ package semver
 
 import (
 	"fmt"
-	"regexp"
-	"strconv"
-	"strings"
+
+	"github.com/Al-Pragliola/go-version"
 )
 
 var (
-	// Link: https://regex101.com/r/Ly7O1x/3/
-	//nolint:lll //We can't wrap regex
-	regex = regexp.MustCompile(`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`)
-
-	ErrInvalidSemver = fmt.Errorf("invalid semantic version")
+	// ErrInvalidVersion is returned when the version is not valid.
+	ErrInvalidVersion = fmt.Errorf("invalid version")
+	// ErrInvalidConstraint is returned when the constraint is not valid.
+	ErrInvalidConstraint = fmt.Errorf("invalid constraint")
 )
 
-// NewVersion takes a string and returns a Version.
-func NewVersion(v string) (Version, error) {
-	if !isValid(v) {
-		return "", fmt.Errorf("%w: %s", ErrInvalidSemver, v)
-	}
-
-	return Version(v), nil
-}
-
-type Version string
-
-func (s Version) String() string {
-	return string(s)
-}
-
-// SamePatchStr takes two version strings and tell if they match down to patch level.
-func SamePatchStr(a, b string) bool {
-	return SamePatch(Version(a), Version(b))
-}
-
-// SamePatch takes two versions and tell if they are part of the same patch.
-func SamePatch(a, b Version) bool {
-	if a == b {
-		return true
-	}
-
-	aMajor, aMinor, aPatch, _ := Parts(a.String())
-	bMajor, bMinor, bPatch, _ := Parts(b.String())
+func GetVersion(v string) (*version.Version, error) {
```

Comment:
not a big deal, but creational functions such as this usually follow the `New<thing>` naming pattern as opposed to `Get<thing>`, as they create a new value object (a version in this case): they don't "take it" from somewhere.